src/providers/SettingContext.tsx

```diff
@@ -0,0 +1,55 @@
+import React, { PropsWithChildren, ReactElement, createContext } from "react"
+import { darkTheme, lightTheme } from "../theme"
+import { CssBaseline } from "@mui/material"
+import { ThemeProvider } from "@mui/material"
+import { getCookie } from "../utils/helper"
+import useMediaQuery from "@mui/material/useMediaQuery"
+
+export type ThemeMode = "light" | "dark" | "system"
```

Comment:
I think we should either use 'system' or null as the type for when the user hasn't explicitly chosen a mode.

Reply:
Updated